src/main/drivers/tim_def.h

```diff
@@ -0,0 +1,357 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <platform.h>
+#include "common/utils.h"
+
+#if defined(STM32F3)
+
+#define DEF_TIM(tim, chan, pin, flags, out) {\
+    tim,\
+    IO_TAG(pin),\
+    EXPAND(DEF_CHAN_ ## chan),\
+    flags,\
+    (DEF_CHAN_ ## chan ## _OUTPUT | out),\
+    EXPAND(GPIO_AF__ ## pin ## _ ## tim ## _ ## chan),\
+    CONCAT(EXPAND(DEF_TIM_DMACHAN_ ## tim ## _ ## chan), _TYPE),\
+    CONCAT(EXPAND(DEF_TIM_DMACHAN_ ## tim ## _ ## chan), _HANDLER)\
+    }
+
+#define DEF_TIM_CHAN(chan) DEF_CHAN_ ## chan 
+#define DEF_TIM_OUTPUT(chan, out) ( DEF_CHAN_ ## chan ## _OUTPUT | out )
+
+#define DEF_DMA_TYPE(tim, chan) CONCAT(EXPAND(DEF_TIM_DMACHAN_ ## tim ## _ ## chan), _TYPE)
+#define DEF_DMA_HANDLER(tim, chan) CONCAT(EXPAND(DEF_TIM_DMACHAN_ ## tim ## _ ## chan), _HANDLER)
+
+/* add the DMA mappings here */
+#define DEF_TIM_DMACHAN_TIM1_CH1    DMA1_CH2
+#define DEF_TIM_DMACHAN_TIM1_CH2    DMA1_CH3
+#define DEF_TIM_DMACHAN_TIM1_CH4    DMA1_CH4
+#define DEF_TIM_DMACHAN_TIM1_TRIG   DMA1_CH4
+#define DEF_TIM_DMACHAN_TIM1_COM    DMA1_CH4
+#define DEF_TIM_DMACHAN_TIM1_UP     DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM1_CH3    DMA1_CH6
+
+#define DEF_TIM_DMACHAN_TIM2_CH3    DMA1_CH1
+#define DEF_TIM_DMACHAN_TIM2_UP     DMA1_CH2
+#define DEF_TIM_DMACHAN_TIM2_CH1    DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM2_CH2    DMA1_CH7
+#define DEF_TIM_DMACHAN_TIM2_CH4    DMA1_CH7
+
+#define DEF_TIM_DMACHAN_TIM3_CH2    DMA_NONE
+#define DEF_TIM_DMACHAN_TIM3_CH3    DMA1_CH2
+#define DEF_TIM_DMACHAN_TIM3_CH4    DMA1_CH3
+#define DEF_TIM_DMACHAN_TIM3_UP     DMA1_CH3
+#define DEF_TIM_DMACHAN_TIM3_CH1    DMA1_CH6
+#define DEF_TIM_DMACHAN_TIM3_TRIG   DMA1_CH6
+
+#define DEF_TIM_DMACHAN_TIM4_CH1    DMA1_CH1
+#define DEF_TIM_DMACHAN_TIM4_CH2    DMA1_CH4
+#define DEF_TIM_DMACHAN_TIM4_CH3    DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM4_UP     DMA1_CH7
+#define DEF_TIM_DMACHAN_TIM4_CH4    DMA_NONE
+
+#define DEF_TIM_DMACHAN_TIM15_CH1   DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM15_CH2   DMA_NONE
+#define DEF_TIM_DMACHAN_TIM15_UP    DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM15_TRIG  DMA1_CH5
+#define DEF_TIM_DMACHAN_TIM15_COM   DMA1_CH5
+
+#ifdef DMA_TIMER16_REMAP 
```

Comment:
You have to remap in sysconfig ... this will pick up the remapping, and apply automatically. Placing this define in the target file will allow this to occur.